content/docs/tech-insights/track-sonarqube/index.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Track On-prem Sonarqube metrics
+publishedDate: '2023-08-04T15:15:00.0Z'
+description: How to track and visualize a Sonarqube metrics with Roadie
+---
+
+One of the most advantageous steps a platform team can take to support application developers (app devs) is to offer them a reliable and secure platform for enhancing code quality. This is often achieved by incorporating SonarCloud into the development process, enabling teams to identify and rectify code-related issues effectively.
+
+By tracking issues and hotspots reported by Sonarqube across your organization allows you to highlight and to proactively address potential pitfalls, enhance code integrity, and bolster application security. This practice not only fosters a culture of continuous improvement but also ensures that software projects evolve with a heightened focus on robustness and reliability.
+
+With Roadie’s Tech Insights feature, you can monitor issues and hotspots right within your software catalog:
+
+1. Automatically scan Components in the Backstage catalog and record the number of issues and hotspots reported by Sonarqube
+2. Visualise the projects which are doing well, and also the worst offenders.
+3. Create a check that can tell us which software is not adhering to standards set by the organization.
+
+As we go through this process, you’ll learn:
+
+1. How to configure a component in Roadie with its Sonarqube project key
+2. How to configure and run the Roadie Sonarqube broker application to connect Sonarqube with Roadie
+4. How to create an HTTP Datasource to collect data from Sonarqube
+5. How to create a Check with Tech Insights
+6. How to create a Scorecard that brings together the Checks with have created in this tutorial
+
+Let’s get started.
+
+## Configure a Component with its Sonarqube project key
+
+1. First identify the project key for the catalog Component.
+You can find this in the Sonarqube dashboard. Find the project, click Administration -> Update key and copy the "Project Key"
+2. Next edit the catalog YAML file for the Component 
+The easiest way to do this is to find the component in the Roadie catalog and click the edit pencil in the top right hand side of the About Card.
+3. Edit and save the YAML file to include the `sonarqube.org/project-key` annotation.
+e.g.
+```yaml
+---
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: test-react-app
+  annotations:
+    sonarqube.org/project-key: punkle_test-react-app
+```
+
+## How to configure and run the Roadie Sonarqube broker application to connect Sonarqube with Roadie
+
+1. First generate a new token in the Sonarqube dashboard
+Click on your profile in the top right of the application, click My Account. Then click Security. Enter a token name and click "Generate Token". Copy the token value.
+2. Run the Roadie broker client application to connect Sonarqube with backstage.
```

Comment:
This doesn't render correctly.

Reply:
I fixed a load of formatting, let me know if it looks better.